daemon/mgr/container.go

```diff
@@ -393,6 +396,38 @@ func (mgr *ContainerManager) Stop(ctx context.Context, name string, timeout time
 	return nil
 }
 
+// Pause pauses a running container.
+func (mgr *ContainerManager) Pause(ctx context.Context, name string) error {
+	var (
+		err error
+		c   *Container
+	)
+
+	if c, err = mgr.container(name); err != nil {
+		return err
+	}
+
+	c.Lock()
+	defer c.Unlock()
+
+	if c.meta.Config == nil || c.meta.ContainerState == nil {
+		err := fmt.Errorf("no container found by %s", name)
+		return httputils.NewHTTPError(err, http.StatusNotFound)
+	}
+
+	if !c.IsRunning() {
```

Comment:
I am wondering what will happen if we are going to pause a paused container?

Reply:
I just run the test on my centos.
If one tries to pause a paused container, the hint goes like "Error: failed to pause container boo: {"message":"container's status is not running: %!d(types.Status=paused)"}".
If one tries to pause a non-existent container, the hint goes like "Error: failed to pause container moo: {"message":"failed to get container info with prefix: moo, more than one"}"